src/VisionModel/resnet.jl

```diff
@@ -0,0 +1,82 @@
+using Flux
+
+basicblock(inplanes::Int, outchannels, downsample::Bool = false) = downsample ? 
+  Chain(Conv((3, 3), inplanes => outchannels[1], stride = 2),
+        BatchNorm(inplanes, λ = relu),
+        Conv((3, 3), inplanes => outchannels[2], stride = 1, pad = 1),
+        BatchNorm(outchannels, λ=relu)) : 
+  Chain(Conv((3, 3), inplanes => outchannels[1], stride = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((3, 3), inplanes => outchannels[2], stride = 1, pad = 1),
+        BatchNorm(outchannels, λ=relu))
+
+bottleneck(inplanes::Int, outchannels, downsample::Bool = false) = downsample ?
+  Chain(Conv((1, 1), inplanes => outchannels[1], stride = 2),
+        BatchNorm(inplanes, λ = relu),
+        Conv((3, 3), inplanes => outchannels[2], stride = 1, pad = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((1, 1), inplanes => outchannels[3], stride = 1),
+        BatchNorm(outchannels, λ = relu)) :
+  Chain(Conv((1, 1), inplanes => outchannels[1], stride = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((3, 3), inplanes => outchannels[2], stride = 1, pad = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((1, 1), inplanes => outchannels[3], stride = 1),
+        BatchNorm(outchannels, λ = relu))
+
+projection(inplanes::Int, outplanes::Int, stride::Int) = Chain(Conv((1, 1), inplanes => outplanes, stride=stride),
+                                                               BatchNorm((outplanes), λ=relu))
+
+identity(inplanes::Int, outplanes::Int, stride::Int) = +
+
+function resnet(block, shortcut_config, channel_config, block_config)
+  layers = []
+  push!(layers, Conv((7, 7), 3=>inplanes, stride=(2, 2), pad=(3, 3)))
+  push!(layers, BatchNorm(inplanes, λ=relu))
+  push!(layers, MaxPool((3, 3), stride=(2, 2), pad=(1, 1)))
+  inplanes = 64
+  baseplanes = 64
+  for nrepeats in block_config
+    outplanes = baseplanes .* channel_config
+    # downsample the first block
+    if shortcut_config == :A
+      # zero padded identity shortcut
```

Comment:
I left comment stubs in my review, but can you fill these out with the actual code? This should be a `SkipConnection` with `identity` shortcut (zero-padding handled within `identity` code).